pages/language/ref/common.mdx

```diff
@@ -1,114 +1,266 @@
-## contractAddressExt
+Usage example:
 
-<Callout type="warning" emoji="⚠️">
-This method throws error if `chain` is invalid or if trying to create address in the masterchain without [masterchain enabled](/language/guides/masterchain).
-</Callout>
+```tact
+let foundMeSome: Address = contractAddress(initOf SomeContract());
+```
+
+## contractAddressExt
 
 ```tact
 fun contractAddressExt(chain: Int, code: Cell, data: Cell): Address;
 ```
-Computes smart contract's Address based on its workchain id `chain`, `code`, `data`.
+
+Computes smart contract's [`Address{:tact}`][p] based on the `chain` it, `code` and `data`. Use [`initOf{:tact}`](/book/expressions#initof) expression to obtain initial `code` and initial `data` of a given contract.
+
+Usage example:
+
+```tact
+let initPkg: StateInit = initOf SomeContract();
+let hereBeDragons: Address = contractAddressExt(0, initPkg.code, initPkg.data);
+```
+
+<Callout type="warning" emoji="⚠️">
+
+  This method throws an error if `chain` is invalid or it points to the masterchain ($-1$) without [masterchain support](/book/masterchain) enabled.
+
+</Callout>
 
 ## emit
 
 ```tact
-fun emit(value: Cell);
+fun emit(body: Cell);
 ```
 
-Sends a message to the outer world with the purpose of logging and analyzing it later off-chain. The message does not have a recipient and is gas-efficient compared to alternatives.
+Sends a message `body` to the outer world with the purpose of logging and analyzing it later off-chain. The message does not have a recipient and is gas-efficient compared to alternatives.
+
+Usage example:
+
+```tact
+emit("Catch me if you can".asComment()); // asComment() converts a String to a Cell
+```
 
 ## cell
 
 ```tact
-fun cell("base64 boc"): Cell;
+fun cell(bocBase64: String): Cell;
 ```
 
-Compile-time function that embeds a Cell into the contract. The Cell is encoded in base64 format.
+A compile-time function that embeds a base64-encoded [BoC](https://docs.ton.org/develop/data-formats/cell-boc#bag-of-cells) `bocBase64` as a [`Cell{:tact}`][p] into the contract.
+
+Usage example:
+
+```tact
+// Init package for Wallet V3R1 as a base64-encoded BoC:
+cell("te6cckEBAQEAYgAAwP8AIN0gggFMl7qXMO1E0NcLH+Ck8mCDCNcYINMf0x/TH/gjE7vyY+1E0NMf0x/T/9FRMrryoVFEuvKiBPkBVBBV+RDyo/gAkyDXSpbTB9QC+wDo0QGkyMsfyx/L/8ntVD++buA=");
+```
 
 ## ton
 
 ```tact
 fun ton(value: String): Int;
 ```
 
-Compile-time function that converts ton's from human-readable format to the integer.
+A compile-time function that converts the given Toncoins `value` from a human-readable format [`String{:tact}`][p] to the nanoToncoins [`Int{:tact}`][int] format.
+
+Usage example:
+
+```tact
+ton("1");           // 10^9 nanoToncoins, which is equal to one Toncoin
+ton("0.1");         // 10^8 nanoToncoins, which is equal to 0.1 Toncoin
+ton("0.000000001"); // 1 nanoToncoin, which is equal to 10^-9 Toncoins
+```
 
 ## dump
+
 ```tact
-fun dump(arg: Any);
+fun dump(arg);
+```
+
+Prints the argument `arg` to the contract's debug console.
```

Comment:
Let's mention one should set things up to actually get it in the console.

Reply:
I know :) It's just sometimes people just read the first paragraph and look at the first code sample :)